src/NuGet.Clients/NuGet.Console/OutputConsole/ChannelOutputConsole.cs

```diff
@@ -0,0 +1,149 @@
+        {
+            return WriteToOutputChannelAsync(_channelId, _channelOutputName, text, CancellationToken.None);
+        }
+
+        private async Task WriteToOutputChannelAsync(string channelId, string displayNameResourceId, string content, CancellationToken cancellationToken)
+        {
+            using (await _pipeLock.EnterAsync())
+            {
+                if (_channelPipeWriter == null)
+                {
+                    var pipe = new Pipe();
+
+                    using (var outputChannelStore = await (await _serviceBrokerClient.GetValueAsync()).GetProxyAsync<IOutputChannelStore>(VisualStudioServices.VS2019_4.OutputChannelStore, cancellationToken))
+                    {
+                        if (outputChannelStore.Proxy != null)
+                        {
+                            await outputChannelStore.Proxy.CreateChannelAsync(channelId, displayNameResourceId, pipe.Reader, TextEncoding, cancellationToken);
+                            _channelPipeWriter = pipe.Writer;
+
+                            // write any deferred messages
+                            foreach (var s in _deferredOutputMessages)
+                            {
+                                // Flush when the original content is logged below
+                                await _channelPipeWriter.WriteAsync(GetBytes(content), cancellationToken);
+                            }
+                            _deferredOutputMessages.Clear();
+                        }
+                        else
+                        {
+                            // OutputChannel is not available so cache the output messages for later
+                            _deferredOutputMessages.Add(content);
+                            return;
+                        }
+                    }
+                }
+                await _channelPipeWriter.WriteAsync(GetBytes(content), cancellationToken);
+                await _channelPipeWriter.FlushAsync(cancellationToken);
+            }
+        }
+
+        private static byte[] GetBytes(string content)
+        {
+            return TextEncoding.GetBytes(content);
+        }
+
+        private async Task CloseChannelAsync()
+        {
+            using (await _pipeLock.EnterAsync())
+            {
+                _channelPipeWriter?.CancelPendingFlush();
+                _channelPipeWriter?.Complete();
+                _channelPipeWriter = null;
+            }
+        }
+
+        public override void StartConsoleDispatcher()
+        {
+            // Nothing to do here in Visual Studio Online Environments mode. We do not create the pane
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!_disposedValue)
+            {
+                if (disposing)
+                {
+                    if (_serviceBrokerClient.IsValueCreated)
+                    {
+                        _serviceBrokerClient.GetValue().Dispose();
```

Comment:
Dispose and WriteToOutputChannelAsync use the _serviceBrokerClient, what will happen if the Dispose is called during a  WriteToOutputChannelAsync method call?

Reply:
Dispose should not throw an exception and the called of write output channel should be ready for a failure. 

I'll take a deeper look into these failure scenarios.

Reply:
Depending on the scenarios, I will let it up to you to decide if the Dispose and WriteToOutputChannelAsync  need to support parallel execution. 
If we decide to not maybe we can add commends to the Dispose method.

Reply:
Added a throw if disposed in every public method. 
Also ensured that no exceptions are thrown from the dispose method (the close method was not guaranteed not to throw).